### PR TITLE
TST/REF: collect Index.equals tests

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/indexes/test_datetimelike.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
+"""
+Tests shared for DatetimeIndex/TimedeltaIndex/PeriodIndex
+"""
+from datetime import datetime, timedelta
+
```
Comment (jreback) [inline pandas/tests/indexes/test_datetimelike.py:5]: should these be in pandas/tests/indexes/datetimelike/test_equals (or as appropriate)?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/test_datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/test_datetimelike.py:5]: havent decided.  im split between implementing tests/indexes/{index_type}/methods/ vs tests/indexes/methods/ for things like equals, leaning towards the latter.

Comment (jreback) [reply pandas/tests/indexes/test_datetimelike.py:5]: yeah doing this for other types, so i think it makes sense here

Comment (jbrockmendel) [reply pandas/tests/indexes/test_datetimelike.py:5]: do you mind holding off on that move while i get a little closer to getting everything collected

Comment (jreback) [reply pandas/tests/indexes/test_datetimelike.py:5]: kk, ok for merge here, but really think this is inconcsistent

**Review by jreback — APPROVED** [review]

**Pull request merged**